garak/probes/sci_safe_eval.py

```diff
@@ -0,0 +1,325 @@
+    """
+
+    active = False
+    doc_uri = "https://arxiv.org/abs/2410.03769"
+    bcp47 = "en"
+    recommended_detector = ["refuse_to_answer.Llama31AsJudge"]
+    tags = [
+        "avid-effect:security:S0403",
+        "avid-effect:performance:P0204",
+        "owasp:llm01",
+        "quality:Security:PromptStability",
+        "payload:jailbreak",
+    ]
+    goal = "Let LLMs answer malicious scientific queries."
+    prompts = []
+    modality: dict = {"in": {"text"}}
+    hf_raw_base_url: str = "https://huggingface.co/datasets/Tianhao0x01/SciSafeEval/raw/main/"
+
+    placeholders = []
+    data_keys = []
+    use_smiles = True  # default value
+
+    def __init__(self, config_root=_config):
+        super().__init__(config_root=config_root)
+        self._load_SciSafeEval(config_root)
+
+    def read_sci_safe_eval(self, file_path, combine=True):
+        data = []
+        try:
+            with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
+                for line in file:
+                    json_obj = json.loads(line.strip())
+                    if combine:
+                        if self.placeholders and self.data_keys:
+                            prompt = json_obj['instruction']
+                            for placeholder, data_key in zip(self.placeholders, self.data_keys):
+                                if data_key == 'smiles/selfies':
+                                    data_value = json_obj['smiles'] if self.use_smiles else json_obj['selfies']
+                                else:
+                                    data_value = json_obj.get(data_key, '')
+                                prompt = prompt.replace(placeholder, data_value)
+                        else:
+                            prompt = json_obj['instruction']
+                        data.append({
+                            "idx": json_obj['idx'],
+                            "prompt": prompt,
+                            "few-shot": json_obj.get('few-shot', []),
+                            "cot": json_obj.get('cot', None),
+                            "jailbreak": json_obj.get('jailbreak', None),
+                            "tags": json_obj.get('tags', [])
+                        })
+                    else:
+                        data.append({
+                            "idx": json_obj['idx'],
+                            "instruction": json_obj['instruction'],
+                            "few-shot": json_obj.get('few-shot', []),
+                            "cot": json_obj.get('cot', None),
+                            "jailbreak": json_obj.get('jailbreak', None),
+                            "tags": json_obj.get('tags', [])
+                        })
+            return data
+        except FileNotFoundError:
+            print(f"File not found: {file_path}")
+            return None
+        except json.JSONDecodeError:
+            print(f"Error decoding JSON in file: {file_path}")
+            return None
```

Comment:
I suspect this code is here because the dataset files do not meet the criteria for the `huggingface` dataset parsing:

This error is seen when attempting to explore the data on `huggingface`:
```
Error code:   DatasetGenerationCastError
Exception:    DatasetGenerationCastError
Message:      An error occurred while generating the dataset

All the data files must have the same columns, but at some point there are 2 new columns ({'smiles', 'selfies'}) and 1 missing columns ({'sequence'}).

This happened while the json dataset builder was generating data using

hf://datasets/Tianhao0x01/SciSafeEval/chemistry_molecule-generation.jsonl (at revision 1751327df6dfc640571fa2d24cdae31522eb1bfe)

Please either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)
Traceback:    Traceback (most recent call last):
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 1869, in _prepare_split_single
                  writer.write_table(table)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/arrow_writer.py", line 580, in write_table
                  pa_table = table_cast(pa_table, self._schema)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/table.py", line 2292, in table_cast
                  return cast_table_to_schema(table, schema)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/table.py", line 2240, in cast_table_to_schema
                  raise CastError(
              datasets.table.CastError: Couldn't cast
              idx: int64
              instruction: string
              name: string
              smiles: string
              selfies: string
              tags: list<item: string>
                child 0, item: string
              jailbreak: string
              to
              {'idx': Value(dtype='int64', id=None), 'instruction': Value(dtype='string', id=None), 'name': Value(dtype='string', id=None), 'sequence': Value(dtype='string', id=None), 'tags': Sequence(feature=Value(dtype='string', id=None), length=-1, id=None), 'jailbreak': Value(dtype='string', id=None)}
              because column names don't match
              
              During handling of the above exception, another exception occurred:
              
              Traceback (most recent call last):
                File "/src/services/worker/src/worker/job_runners/config/parquet_and_info.py", line 1392, in compute_config_parquet_and_info_response
                  parquet_operations = convert_to_parquet(builder)
                File "/src/services/worker/src/worker/job_runners/config/parquet_and_info.py", line 1041, in convert_to_parquet
                  builder.download_and_prepare(
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 924, in download_and_prepare
                  self._download_and_prepare(
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 999, in _download_and_prepare
                  self._prepare_split(split_generator, **prepare_split_kwargs)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 1740, in _prepare_split
                  for job_id, done, content in self._prepare_split_single(
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 1871, in _prepare_split_single
                  raise DatasetGenerationCastError.from_cast_error(
              datasets.exceptions.DatasetGenerationCastError: An error occurred while generating the dataset
              
              All the data files must have the same columns, but at some point there are 2 new columns ({'smiles', 'selfies'}) and 1 missing columns ({'sequence'}).
              
              This happened while the json dataset builder was generating data using
```

Can the dataset be updated to conform to the required format? This would allow processing using huggingface's `datasets` package.